Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Cancel.hs

```diff
@@ -45,17 +45,14 @@ cancel ::
 cancel transporterId subscriber req =
   withFlowHandlerBecknAPI . withTransactionIdLogTag req $ do
     logTagInfo "Cancel API Flow" "Reached"
-    dCancelReq <- ACL.buildCancelReq req
-    case dCancelReq of
-      Left cancelReq -> do
-        Redis.whenWithLockRedis (cancelLockKey cancelReq.bookingId.getId) 60 $ do
-          (merchant, booking) <- DCancel.validateCancelRequest transporterId subscriber cancelReq
-          fork ("cancelBooking:" <> cancelReq.bookingId.getId) $
-            DCancel.cancel cancelReq merchant booking
-      Right cancelSearchReq -> do
-        searchReq <- DCancel.validateCancelSearchRequest transporterId subscriber cancelSearchReq
-        fork ("cancelSearch:" <> cancelSearchReq.transactionId) $
```

Comment:
Why did you remove the fork here? And why to revert all these changes they were done to reduce the latency, after you refactoring it will be the same code it was before.

Reply:
> Why did you remove the fork here? And why to revert all these changes they were done to reduce the latency, after you refactoring it will be the same code it was before.

I did not remove it, [i moved it in domain logic](https://github.com/nammayatri/nammayatri/pull/985/files/3eae37014dc8846d5372658f176d6ad6021dc87a#diff-3359acbc2958c5b0634d3cdb45da8ffbd7be4674fc0e0a2c36bc08db9c0fb24fR88). It's much easier to work with that way, no redundant data types.